add task 8

nktrn opened the pull request:
Информация про посты которые хранятся в коллекции:
Количество постов: 20
Авторы и сколько раз встречаются: Sasha - 10, Katya - 4, Nikita - 6
Хэштэги: weather, sun , rain, NewYear, snow, street, clouds 
Я не очень понял как сделать этот проект на основе 5 задания, поэтому я создал новый.
------
Первым делом надо добавить эти посты в список, для этого нужно ввести в адресную строку http://localhost:8080
Посты хранятся в классе PostArrayList.java. В этом классе есть статическая переменная list, где хранятся посты.
В конструкторе этого класса передается строка, которая хранит в себе информация о постах. 
Я реализовал класс JsonParser со статическими методами, которые парсят json в объект Post, ArrayList<Post> и наоборот из Post в json. С помощь этого класса в конструкторе класса PostArrayList я создаю список из постов.
Строка с информацией о постах хранится в index.jsp и при переходе на страницу http://localhost:8080 создается list. В index.jsp так же подключил файл index.html, javaScript и css со всеми нужными файлами хранятся в папке webapp/resources. По этому вроде бы все.
Класс Post - в нем хранится вся информация о посте. Дату я храню в String, у меня не получилось хранить ее в LocalDateTime, потому что когда парсил json вылетали ошибки и не знал как это исправить, мучался долго но не получилось. 
Однако в методе getCreatedAt порсю createdAt в LocalDateTime и возвращаю уже дату а не строку, только так получилось.
Так же в этом классе реализованы различные геттеры и сеттеры а так же проверка поста на валидатность: если поля id, author, createdAt и одно из: descriprion или photoLink не пустые, тогда пост валидный.
------
Чтобы получить пост с определенным id, нужно ввести в адресную строку http://localhost:8080/tweets и указать параметр id(http://localhost:8080?id=2). Если пост с указанным id существует, то в браузере выведется информация о нем, иначе выведется null. Для того чтобы узнать существует ли пост с указанным id в классе PostArrayList реализован метод postWithIdExist(String id) который возвращает true если существует и false в противном.
Если существует то вызываю метод в PostArrayList под названием getPostById(String id) которые возвращает пост с id.
Так же можно получить список постов с какогото номера по какой то. http://localhost:8080/tweets?from=x&to=y.
Если from и to указаны верно то в браузере выведется список постов начиная с номера from до номера.
Удаление поста происходит аналогично получению поста по id. Если пост с указанным id существует то в браузере выведется то что пост был удален успешно, если не существует то выведется то что поста с таким id не существует.
------
чтобы отредактировать какой либо пост, я в postmane по адресу http://localhost:8080/tweets/edit передавал json файл, который парсил в объект Post и заменял параметры descriprion, hashTags и photoLink в списке постов с таким же id.
Если в json будет отсуствовать какой либо из этих трех параметров, то они заменятся на null после редактирования поста.
В json главное передать id поста которых нужно отредактировать.
На экран выводится сам пост до редактирования, является ли он валидатным до редактирования, пост после редактирования и является ли он
валидатным после редактирования.
Если поста с переданным id не существует выведется соответствующая информация.
-------
http://localhost:8080/tweets/search
Параметры:author, h(хэштеги) их передвать надо через строку вот так http://localhost:8080/tweets/search?h=weather%20rain%20pain - это пример
Если два параметра не указаны то выведется все посты отсортированные по дате создания.
Если указан author то выведятся все посты отсортированные по дате создания с данным автором
Если указан h, то выведятся все посты с содержащие хэштеги. Параметр h преобразовывается в массив и ищутся те посты, в которых есть эти хештеги. Если такие посты есть, то они выводятся отсортированные по дате создания.
Сделал так же сортировку когда эти два параметра передаются оба, но она почему то не работает, буду пытаться доделать.
-------
http://localhost:8080/tweets/add - добавление поста.
в postmane передавал json. Он парсится в объект Post и дальше смотрится: если пост валидный и поста с таким же id не существует в списке постов, то он добавляется, если нет то выводится соответствующая информация.
------
Столкнулся с проблемой, раньше когда у меня в html в роли лайка выступал смайлик, и все работало нормально,
но когда перенес html в идею там вместо него и пары других символов появились какие то иероглифы
------
Я так же уже давно отправил 5 задание, сейчас доделал его, но у меня почему то не получается закоммитить и отправить на github.

nktrn commented:
ой, там случайно добавилось UI с прошлых заданий

smiakchilo commented on Lenter/target/Lenter/WEB-INF/web.xml:
```diff
@@ -0,0 +1,47 @@
+    <servlet-name>index.jsp</servlet-name>
+    <jsp-file>/index.jsp</jsp-file>
+  </servlet>
+  <servlet-mapping>
```
This mapping is redundant. Anything named "index" is processed by default

smiakchilo commented on Lenter/target/Lenter/index.jsp:
```diff
@@ -0,0 +1,293 @@
+    <title>Lenter</title>
+</head>
+<body>
+    <%String str = "[\n" +
```
Could not quite  understand what's going on here, but anyway this form of notation is inacceptable. No one writes json like that. Are you writing it down to just parse in to Posts list? So why not create a posts list as it is? 
And also I haven't caught the idea how you gonna use this list in the premises of this page

smiakchilo commented on Lenter/target/Lenter/resources/script.js:
```diff
@@ -0,0 +1,322 @@
+//Количество постов: 20
```
You should avoid Russian-language comments whatever you write

nktrn commented:
so, i'll try to speak english. I forgot to add some files to .gitignore, sorry:(
i removed dependencies that are not used,  changed the java version to 1.8.
Renamed some methods which you pointed to, remove variable like which was responsible for the number of likes and now type of variable of id is int. Why i made list static? Because 
then it will save posts until i terminate or redepploy my server, for me it was eathier  to test 
different  mehods. And i made an intiallization of this list in index.jsp, so it will fill up when we
at the first time visiting page. Also why this json string is so crazy, just becouse im lazy and did not 
made new json and copied it from task7 and then cntl c+v. 
Also i added org.apache.commons.lang3 dependency and changed method validated. Realy this is cool thing, thank you for it, also with this thing I simplified another one method.
Added in method getPostByauthor check whether the author is not null and simplified  sort.
log messages in response and empty overrides i removed.
And last, i made index.jsp mapping because the browser loads in my project index.html instead.
I hope my English was not very terrible
